mmv1/third_party/terraform/services/backupdr/data_source_backup_dr_backup_plan_association.go

```diff
@@ -44,3 +45,116 @@ func dataSourceGoogleCloudBackupDRBackupPlanAssociationRead(d *schema.ResourceDa
 	}
 	return nil
 }
+
+// Plural datasource to Fetch BackupPlanAssociations for a given resource type
+func DataSourceGoogleCloudBackupDRBackupPlanAssociations() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceGoogleCloudBackupDRBackupPlanAssociationsRead,
+		Schema: map[string]*schema.Schema{
+			"location": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The location to list the backup plan associations from.",
+			},
+			"project": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "The ID of the project in which the resource belongs.",
+			},
+			"resource_type": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `The resource type of workload on which backup plan is applied. Examples include, "compute.googleapis.com/Instance", "compute.googleapis.com/Disk".`,
+			},
+
+			// Output: a computed list of the associations found
```

Comment:
can you clean up a few of the extraneous comments like this, and only leave ones that point out unexpected logic or logic that may deviate from normal patterns? Too many distract from the more critical comments in my opinion